File: application/src/main/javascript/workspace/projects/shared/src/lib/domain/descriptors/validators/PatternValidator.ts
import Validator from "./Validator";
import Entity from "../../../mapper/annotations/Entity";
import Primitive from "../../../mapper/annotations/Primitive";
import { AsControl } from "../../../directives/as-control";

@Entity("PatternValidator")
export default class PatternValidator implements Validator<string> {

    @Primitive()
    regexp : string

    constructor(regexp: string) {
        this.regexp = regexp;
    }

    validate(control: AsControl<string>): boolean {
        let regex = new RegExp(this.regexp, "g")
        return regex.test(control.model())
    }

}

File: library/json-mapper2/src/main/scala/com/anjunar/scala/schema/model/validators/PatternValidator.scala
package com.anjunar.scala.schema.model.validators

import com.anjunar.scala.mapper.annotations.{IgnoreFilter, PropertyDescriptor}

import scala.annotation.meta.field

@IgnoreFilter
case class PatternValidator(@(PropertyDescriptor @field)(title = "Regex") regex: String) extends Validator
